roles/assemble_template/tasks/main.yml

```diff
@@ -36,6 +38,7 @@
     loop_var: __fragment
     label: "{{ __fragment.path | basename }}"
   when: fragments.matched > 0
+  delegate_to: localhost
```

Comment:
I believe this is redundant; `ansible.builtin.template` only runs on the controller.

Reply:
Ah, I see, we need to ensure the `dest` is on `localhost` as well.